pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -661,6 +682,18 @@ func (r *ReconcileComplianceScan) phaseAggregatingHandler(h scanTypeHandler, log
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileComplianceScan) fetchResultCounts(scan *compv1alpha1.ComplianceScan, logger logr.Logger) (int, error) {
+	var checkList compv1alpha1.ComplianceCheckResultList
+	checkListOpts := client.MatchingLabels{
+		compv1alpha1.ComplianceScanLabel: scan.Name,
+	}
+	if err := r.Client.List(context.TODO(), &checkList, &checkListOpts); err != nil {
+		logger.Error(err, "Cannot list the check results")
```

Comment:
Here we could supply additional information about the exact scan we couldn't obtain a `ComplianceCheckResultList` for, making it more informative in the logs.

Reply:
I think it makes sense to have additional information, but I wonder if adding that might be unnecessary because https://github.com/ComplianceAsCode/compliance-operator/pull/588/files/cc9a0d89b221a015818a2ddd026caf49e9220fc0#diff-7aba53b74d27b417a478166b1a983b145fb0e381975e97097b7794752b51edeeR134 already logging with the request

Reply:
example of log
`{"level":"info","ts":"2024-08-20T11:20:48.127Z","logger":"scanctrl","msg":"Reconciling ComplianceScan","Request.Namespace":"openshift-compliance","Request.Name":"e2e-moderate"}
`

Reply:
```
{"level":"error","ts":"2024-08-20T11:20:48.166Z","logger":"scanctrl","msg":"Cannot update the status","Request.Namespace":"openshift-compliance","Request.Name":"e2e-moderate","error":"Operation cannot be fulfilled on compliancescans.compliance.openshift.io \"e2e-moderate\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"github.com/ComplianceAsCode/compliance-operator/pkg/controller/compliancescan.(*ReconcileComplianceScan).phasePendingHandler\n\tgithub.com/ComplianceAsCode/compliance-operator/pkg/controller/compliancescan/compliancescan_controller.go:310\ngithub.com/ComplianceAsCode/compliance-operator/pkg/controller/compliancescan.(*ReconcileComplianceScan).Reconcile\n\tgithub.com/ComplianceAsCode/compliance-operator/pkg/controller/compliancescan/compliancescan_controller.go:189\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:114\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:311\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:261\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:222"}
```

Reply:
Aha - good call. Thanks for clarifying.